wave_lang/kernel/wave/codegen/handlers.py

```diff
@@ -596,7 +605,33 @@ def handle_generic_binary(emitter: WaveEmitter, node: fx.Node):
                     f"rhs={get_custom(op.rhs)}"
                 )
 
-            result = binary_fn(lhs, rhs, emitter.options)
+            if (
+                maybe_scalarize
+                and emitter.options.scalarize_packed_math
+                and _near_mma(node)
+            ):
+                lhs_values = []
+                for i in range(lhs.type.shape[0]):
+                    lhs_elem = vector_d.extract(
+                        lhs, static_position=[i], dynamic_position=[]
+                    )
+                    lhs_values.append(lhs_elem)
+                rhs_values = []
+                for i in range(rhs.type.shape[0]):
+                    rhs_elem = vector_d.extract(
+                        rhs, static_position=[i], dynamic_position=[]
+                    )
+                    rhs_values.append(rhs_elem)
+                values = []
+                for lhs_elem, rhs_elem in zip(lhs_values, rhs_values):
+                    result = binary_fn(lhs_elem, rhs_elem, emitter.options)
+                    values.append(result)
```

Comment:
Can we apply loop fusion here?

Reply:
I specifically dit it this way so in IR we have nice
```
vector.extract
vector.extract
vector.extract
vector.extract
vector.extract
vector.extract
vector.extract
vector.extract
arith.fadd
arith.fadd
arith.fadd
arith.fadd
```
Instead of mix of them

Reply:
how about we only loop fusion the lhs and rhs? haha